pkg/yamltemplate/filetests/ytt-library/assert/assert-equals.yml

```diff
@@ -0,0 +1,18 @@
+#@ load("@ytt:assert", "assert")
+
+equals_works_strings: #@ assert.equals("foo", "foo")
+equals_works_numbers: #@ assert.equals(1, 1)
+equals_works_deciaml: #@ assert.equals(3.14159, 3.14159)
+equals_works_boolean: #@ assert.equals(True, True)
+equals_works_none: #@ assert.equals(None, None)
+equals_works_array: #@ assert.equals([1, 2], [1, 2])
+equals_works_map: #@ assert.equals({"foo": "bar"}, {"foo": "bar"})
++++
```

Comment:
🎉  for adding maps and arrays!

 One more test I would like to see is for [`yamlFragments`](https://carvel.dev/ytt/docs/latest/lang-ref-yaml-fragment/#docs)

Reply:
Got a test working for the yamlfragment failure case  - The TestYAMLTemplate function is _extremely_ picky about whitespace and it was exceedingly frustrating to get a test to pass.

This is what the error message looks like (or at least gets the test to pass)
```
- assert.equals: yamlfragments are not equal:
    in <toplevel>
      stdin:12 | test_equals: #@ assert.equals(yamlFunc(), moreYamlFunc())

    reason:
     Expected:
     foo: bar
     -----
     Actual:
     bar: foo
```

Reply:
This is a good format. For consistency in this codebase let's follow the testing library we use elsewhere. They present the information in the format below. This does not have capitalization, no leading whitespace, and a newline to separate the two blocks:

https://github.com/stretchr/testify/blob/ab6dc3262822ed562480c19876b0257ace761e3e/assert/assertions.go#L346-L348

To introduce as few new concepts in the error message as possible, let's leave out the word `yamlFragment`.